### PR TITLE
Unit tests for esri.core services

### LINKED ISSUE #146: Remaining esri.core unit tests
Not planning to unit test directives as we have excellent coverage via protractor tests

- [x] core/esriLoader (_some tests already started_)
- [x] core/esriMapUtils
- [x] core/esriRegistry
- [x] core/esriLayerUtils

### PR DESCRIPTION
**Pull request by tomwayson**
Resolves #146 

Couldn't get tests to work for `esriMapUtils.createWebMap()`, but those can be added later. Otherwise I'm happy w/ the tests for the loader and map and layer utils.

@jwasil please review and merge at will.

### CODE REVIEW

**Comment by jwasil** [comment]
@tomwayson reviewing now and will push some stuff soon. You'll love the "fix" to get createWebMap tests running.

File: test/unit/core/esriMapUtils.spec.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                        'wkid': 102003
+                    }
+                }]);
+            });
+            $rootScope.$digest();
+        });
+    });
+
+    // TODO: tests for create web map not working
+    xdescribe('createWebMap', function() {
```
Comment (jwasil) [inline test/unit/core/esriMapUtils.spec.js]: @tomwayson was **x**describe on purpose to stop these tests from running?  I played around with these and also couldn't get the actual `expect`s to show up in the test runners either way.

Comment (tomwayson) [inline test/unit/core/esriMapUtils.spec.js]: Yeah. Just skipping this until we figure out the issue. Also, I was thrashing around a bit trying different ways of getting it working and it doesn't even look like I checked in the code I expected to work.

Comment (jwasil) [inline test/unit/core/esriMapUtils.spec.js]: Understood, thanks.  Safe to go ahead and merge, or does that mean you might want to add some more?

**Pull request merged**